errors.go

```diff
@@ -70,3 +70,17 @@ func (t TypeError) Error() string {
 func NewTypeError(expected, actual interface{}) error {
 	return TypeError{fmt.Sprintf("Invalid type for arguments. Expected: %T, actual: %T", expected, actual)}
 }
+
+type CanceledError struct {
+	p string
```

Comment:
They probably should be. I can see sometimes baking in data that you might want to retrieve on the other side, usually with custom error interfaces. But if you only ever need to retrieve the string, there's no point to the error being typed anything other than that.